export_snowflake/__init__.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+
+import argparse
+import io
+import json
+import traceback
+import logging
+import os
+import sys
+import boto3
+import time
+
+from singer import get_logger
+from datetime import datetime, timedelta
+
+from export_snowflake.file_formats import csv
+from export_snowflake.file_formats import parquet
+from export_snowflake import stream_utils
+
+from export_snowflake.db_sync import DbSync
+from export_snowflake.file_format import FileFormatTypes
+from export_snowflake.exceptions import (
+    SymonException
+)
+
+LOGGER = get_logger('export_snowflake')
+
+# Tone down snowflake.connector log noise by only outputting warnings and higher level messages
+logging.getLogger('snowflake.connector').setLevel(logging.WARNING)
+
+# for symon error logging
+ERROR_START_MARKER = '[target_error_start]'
+ERROR_END_MARKER = '[target_error_end]'
+
+LOCAL_SCHEMA_FILE_PATH = 'local_schema.json'
+
+def get_snowflake_statics(config):
+    """Retrieve common Snowflake items will be used multiple times
+
+    Params:
+        config: configuration dictionary
+
+    Returns:
+        tuple of retrieved items: table_cache, file_format_type
+    """
+    db = DbSync(config)
+    # The file format is detected at DbSync init time
+    file_format_type = db.file_format.file_format_type
+    
+    return file_format_type
+
+def direct_transfer_data_from_s3_to_snowflake(config, o, file_format_type):    
+    # retrieve the schema file from s3 bucket
+    # if the table doesn't exist, create the new table based on the schema
+    s3_client = boto3.client('s3')
+
+    bucket = config["bucket"]
+    prefix = config["prefix"]
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+
+    schema_file = None
+    for obj in response['Contents']:
+        # for both python and spark parquet_to_csv steps, we always have a schema.json with the schema details
+        if obj['Key'].endswith('.json'):
+            schema_file = obj['Key']
+    
+    if schema_file is None:
+        raise Exception("Schema file not found in the s3 bucket")
+
+    s3_client.download_file(bucket, schema_file, LOCAL_SCHEMA_FILE_PATH)
+
+    f = open(LOCAL_SCHEMA_FILE_PATH)
+    schema = json.load(f)
+
+    # generate the schema object for DbSync
+    stream = config["stream"]
+    o = {
+        "stream": stream,
+        "schema": {"properties": schema['fields']},
+        "key_properties": config["key_columns"]
+    }
+    
+    db_sync = DbSync(config, o, None, file_format_type)
+
+    try:
+        transfer_start_time = time.time()
+
+        # check if the schema(Snowflake table) exists and create it if not
+        db_sync.create_schema_if_not_exists()
+        
+        # sync_table will check the schema with the table in Snowflake, if there's a miss matching in columns, raise an error
+        db_sync.sync_table()
```

Comment:
In sync_table, we handle OWNERSHIP privilege on table. I think it was needed to perform ddl + using table stage. With new approach, do we still need table ownership? If not, error msg might be misleading. Do you have a new list of permissions required for new approach? I think we might need to update documentation with new set of permissions needed in the future.